File: SurperPm-agent-web/frontend/src/lib/schemas/discussion.ts
import { z } from "zod";

export const discussionSchema = z.object({
  id: z.number(),
  workspace_id: z.string().optional().default(""),
  goal_id: z.number().nullable().optional(),
  role: z.string().default("user"),
  content: z.string().default(""),
  author: z.string().nullable().optional(),
  parent_id: z.number().nullable().optional(),
  topic_id: z.number().nullable().optional(),
  image_data_uri: z.string().nullable().optional(),
  card_response: z.any().nullable().optional(),
  created_at: z.string().optional().default(""),
});

export type Discussion = z.infer<typeof discussionSchema>;

export const discussionListSchema = z.array(discussionSchema);

File: SurperPm-agent-web/backend/app/models/discussion.py
"""Discussion table model."""

from datetime import UTC, datetime

from sqlmodel import Field, SQLModel


class Discussion(SQLModel, table=True):
    __tablename__ = "discussion"

    id: int | None = Field(default=None, primary_key=True)
    workspace_id: str = Field(foreign_key="workspace.id", nullable=False)
    goal_id: int | None = Field(default=None, foreign_key="goal.id")
    role: str = Field(nullable=False)  # user | agent | system
    content: str = Field(nullable=False)
    author: str | None = Field(default=None)
    parent_id: int | None = Field(default=None, foreign_key="discussion.id")
    topic_id: int | None = Field(default=None, foreign_key="topic.id")
    created_at: datetime = Field(default_factory=lambda: datetime.now(UTC))
